skrub/_agg_joiner.py

```diff
@@ -0,0 +1,592 @@
+            the join operation.
+
+        columns_to_agg : str or array-like. Optional, can be missing.
+            Select the columns from the auxiliary dataframe to use as values during
+            the aggregation operations.
+            If missing, use all columns except `columns_to_join`.
+
+    main_key : str or array-like
+        Select the columns from the main table to use as keys during
+        the join operation.
+
+        If main_key refer to a single column, it will be used to join all tables.
+
+        Otherwise, if main_key is a list it must specify the key for each table
+        to join.
+
+    agg_ops : str or list of str, default=None
+        Aggregation operations to perform on the auxiliary table.
+
+        numerical : {"sum", "mean", "std", "min", "max", "hist", "value_counts"}
+
+        categorical : {"mode", "count", "value_counts"}
+
+        If set to None, ['mean', 'mode'] will be used.
+
+    suffixes : list of str, default=None
+        The suffixes that will be add to each table columns in case of
+        duplicate column names, similar to `("_x", "_y")` in :obj:`~pandas.merge`.
+
+        If set to None, we will use the table index, by looking at their
+        order in 'tables', e.g. for a duplicate columns: price (main table),
+        price_1 (auxiliary table 1), price_2 (auxiliary table 2), etc.
+
+    See Also
+    --------
+    :class:`AggTarget` :
+        Aggregates the target `y` before joining its aggregation
+        on the base dataframe.
+    :class:`FeatureAugmenter` :
+        Augments a main table by automatically joining multiple
+        auxiliary tables on it.
+
+    Examples
+    --------
+    >>> main = pd.DataFrame({
+            "userId": [1, 1, 1, 2, 2, 2],
+            "movieId": [1, 3, 6, 1, 3, 6],
+            "rating": [4.0, 4.0, 4.0, 3.0, 2.0, 4.0],
+            "genre": ["drama", "drama", "comedy", "sf", "comedy", "sf"],
+        })
+    >>> join_agg = AggJoiner(
+            tables=[
+                (main, "userId", ["rating", "genre"]),
+                (main, "movieId", ["rating"]),
+            ],
+            main_key=["userId", "movieId"],
+            suffixes=["_user", "_movie"],
+            agg_ops=["mean", "mode"],
+        )
+    >>> join_agg.fit_transform(main)
+        userId  movieId  rating   genre genre_mode_user  rating_mean_user  rating_mean_movie
+    0       1        1     4.0   drama           drama               4.0   3.5
+    1       1        3     4.0   drama           drama               4.0   3.0
+    2       1        6     4.0  comedy           drama               4.0   4.0
+    3       2        1     3.0      sf              sf               3.0   3.5
+    4       2        3     2.0  comedy              sf               3.0   3.0
+    5       2        6     4.0      sf              sf               3.0   4.0
+    """  # noqa: E501
+
+    def __init__(self, tables, main_key, agg_ops=None, suffixes=None):
```

Comment:
Using keyword arguments could make readability way better:
```suggestion
    def __init__(self, *, tables, main_key, agg_ops=None, suffixes=None):
```

Reply:
Yes, but it also forces people to use kwargs, so I'm hesitant.

Reply:
scikit-learn, numpy, scipy or pandas objects don't behave like this, so I would weight that against enforcing good practices that might be misleading for new users. 

Since the scope of this is broader than this PR, let's wait for a concensus.

Reply:
You're right, pandas & co don't, that's the reason we didn't enforce that in the `fuzzy_join`, since we want it to be a drop-in replacement of `pandas.merge`. But I don't think this logic applies here, and similar to the [other encoders](https://github.com/skrub-data/skrub/blob/3e3075c5812be45d2e21dc312c8d5f58623dff45/skrub/_gap_encoder.py#L650) and even the [`Joiner`](https://github.com/skrub-data/skrub/blob/2473cd8ffb626a67c9a8275a922021393e3fee80/skrub/_joiner.py#L129), I think this should be the norm. For reference, #514.

I agree that this should not be blocking this PR, but definitely something that needs to be discussed and addressed before the release :)